core/src/main/java/com/klaytn/caver/contract/Contract.java

```diff
@@ -0,0 +1,349 @@
+package com.klaytn.caver.contract;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.klaytn.caver.Caver;
+import com.klaytn.caver.abi.ABI;
+import com.klaytn.caver.methods.request.KlayLogFilter;
+import com.klaytn.caver.methods.response.Bytes32;
+import com.klaytn.caver.methods.response.KlayLogs;
+import com.klaytn.caver.methods.response.TransactionReceipt;
+import com.klaytn.caver.transaction.response.PollingTransactionReceiptProcessor;
+import com.klaytn.caver.transaction.response.TransactionReceiptProcessor;
+import com.klaytn.caver.transaction.type.SmartContractDeploy;
+import com.klaytn.caver.utils.CodeFormat;
+import com.klaytn.caver.utils.Utils;
+import io.reactivex.Flowable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Consumer;
+import org.web3j.abi.datatypes.Type;
+import org.web3j.protocol.ObjectMapperFactory;
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.methods.response.EthSubscribe;
+import org.web3j.protocol.exceptions.TransactionException;
+import org.web3j.protocol.websocket.events.LogNotification;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Contract {
+
+    /**
+     * A caver instance.
+     */
+    Caver caver;
+
+    /**
+     * A contract ABI(Application Binary interface) json string.
+     */
+    String abi;
+
+    /**
+     * A contract address.
+     */
+    String contractAddress;
+
+    /**
+     * The map where method name string and ContractMethod mapped.
+     */
+    Map<String, ContractMethod> methods;
+
+    /**
+     * The map where event name string and ContractEvent mapped.
+     */
+    Map<String, ContractEvent> events;
+
+    /**
+     * The ContractMethod instance related Contract's constructor.
+     */
+    ContractMethod constructor;
+
+    /**
+     * Creates a Contract instance.
+     * @param caver A Caver instance.
+     * @param abi A contract's ABI(Application Binary interface) json string.
+     * @throws IOException
+     */
+    public Contract(Caver caver, String abi) throws IOException{
```

Comment:
Should the abi parameter be String type? What should the user do to use the result of compiling with solc?

Reply:
The ABI string in test code is also JSON formatted string. It just add carriage return("\n").
If user use compile .sol file through solc compiler, user can get .abi and .bin files. 
User can read contents in this file and make a String object.